src/main/java/io/redisearch/client/Client.java

```diff
@@ -488,10 +488,17 @@ private BinaryClient addDocument(Document doc, AddOptions options, Jedis conn) {
         }
 
         args.add(Keywords.FIELDS.getRaw());
-        for (Map.Entry<String, Object> ent : doc.getProperties()) {
-            args.add(SafeEncoder.encode(ent.getKey()));
-            Object value = ent.getValue();
-            args.add(value instanceof byte[] ?  (byte[])value :  SafeEncoder.encode(value.toString()));
+        String key = null;
+        try {
+            for (Map.Entry<String, Object> ent : doc.getProperties()) {
+                key = ent.getKey();
+                args.add(SafeEncoder.encode(key));
+                Object value = ent.getValue();
+                args.add(value instanceof byte[] ?  (byte[])value :  SafeEncoder.encode(value.toString()));
```

Comment:
I thought perhaps just add here a check if the value is null. 
And `throw new NullPointerException("Document attribute '"+ key +"' is null. (Remove it, or set a value)" )`

Reply:
Done